app/models/concerns/course/levels_concern.rb

```diff
@@ -0,0 +1,52 @@
+module Course::LevelsConcern
+  extend ActiveSupport::Concern
+
+  # Generates the level numbers for the Course::Levels,
+  # since only the thresholds are stored in database.
+  # The entries should have been sorted during the
+  # database query.
+  #
+  # @param [CollectionProxy] levels_collection Contains Course::Levels
+  # @return [Array<Course::Levels>] Array with numbered Course::Levels
+  def number_levels(levels_collection)
+    levels_array = levels_collection.each_with_index.map do |level, index|
```

Comment:
No need to assign to a variable.

Are you not directly referencing the course's levels?